.devops/azure-templates/setup-project.yml

```diff
@@ -0,0 +1,39 @@
+# Azure DevOps pipeline template used to setup the Node project:
+# 1. checkout code
+# 2. setup correct node version
+# 3. install node dependencies
+
+parameters:
+- name: 'cache_version_id'
+  type: string
+  default: $(CACHE_VERSION_ID)
+
+# the branch, tag or commit to deploy
+- name: 'gitReference'
+  type: string
+  default: '$(Build.SourceVersion)'
+  
+steps:
+- checkout: self
+  displayName: 'Checkout'
+
+# This is needed because the pipeline may point to a different commit than expected
+# The common case is when the previous stage pushed another commit
+- ${{ if ne(parameters.gitReference, variables['Build.SourceVersion']) }}:
+  - script: |
+      git fetch && git checkout ${{ parameters.gitReference }}
+    displayName: 'Checkout reference'  
+
+- task: UseNode@1
+  inputs:
+    version: $(NODE_VERSION)
+  displayName: 'Set up Node.js'
+
+- script: |
+    echo registry=https://registry.npmjs.com/ >> .npmrc
+    echo \//npm.pkg.github.com/:_authToken=$GITHUB_TOKEN >> .npmrc
+    echo \@pagopa:registry=https://npm.pkg.github.com/ >> .npmrc
```

Comment:
Can we use the `scripts/init-npm.sh` script file?

Reply:
I was actually thinking on removing such script... Do we use it other than in pipelines?

Take in account that it would be no longer needed as soon as `@pagopa/io-spid-commons` is moved to a public registry (which is happening while we discuss it).

Reply:
This will be updated in #762